### PR TITLE
Spatial Interdictor upgrade: alternate mainboards and balance/code pass

### PR DESCRIPTION
**Pull request by Kubius**
[FEAT][OBJECTS]
## About the PR 

Major update to spatial interdictors, primarily centered on the introduction of new mainboards granting them secondary functionality.

**Alternate Mainboards**

The interdictor manufacture Manudrive now provides a discrete blueprint for mainboard manufacture (as opposed to bundling it with the frame kit), with four options for mainboards to produce. Each board has a slightly different cost. Their installation affects the class of the interdictor, sometimes introducing alternate functionality.

- **Standard interdictors** (no bonus functionality) will receive no special class designation, but receive one extra unit of radius on top of the radius provided by the installed phase-control rod. Sigma phase-control rods have been reduced from 7 radius to 6, to keep the top-end radius the same.
- **Nimbus-class interdictors** can wirelessly charge cyborgs, streaming power directly from the interdictor to the cyborg's internal cell. Each time the cyborg's power use ticks, and the cyborg's cell power is below the recharge packet size (currently 75 cell units of power), any Nimbus-class interdictor in range will supply the cyborg with power if capable of doing so.
- **Zephyr-class interdictors** create a bioelectric field around them, carefully calibrated for compatibility with human bioelectrics. When standing within range of a Zephyr-class interdictor, you'll get a Zephyr Field status effect that supplements your natural stamina regeneration; the effect can linger briefly after leaving the field if it's had time to accumulate, though not for more than a few seconds.
- **Devera-class interdictors** use a combination of ionization and micro-vibrations to suppress and deter bacteria and other "funk" that's exposed in the vicinity. Miasma emission from rotting corpses is suppressed, and when motives are enabled, passive degradation of hygiene is eliminated (does not *increase* hygiene, just prevents it from dropping unless you outright get gunk on you).

**Other Changes**

- **Mass radiation protection has been dramatically improved.** The cost per life tick to protect one person has gone from 250 cell units to 60, and is now mult-adjusted; this should now allow a fully charged 15,000 unit cell to shield three people through the full duration without charging, or about a dozen people with adequate grid power.
- The expend_interdict proc has been expanded to include inbuilt handling of range detection and interdictor type comparison. This minimizes repeated code, is likely a bit more performant, and is simpler to set up.
- Interdiction check loops have been globally standardized to use for_by_tcl.
- Adjusted some "annoyance bits" with the interdictor; visible messages in chat have been cut from assembly steps, and interdiction noise volume now varies based on the amount of power expended to perform the interdiction.
- Manually spawned interdictors will now start charged, with an interdict range of 5. Subvariants are included for each special mainboard type.

## Why's this needed? 

Allowing interdictors to be constructed with supplementary secondary functions provides an additional "value add" with the potential to be useful even if the random events never roll around. Radstorm improvement 

## Changelog 

```changelog
(u)Kubius
(*)Spatial interdictors may now be constructed with alternate control boards that grant them new functionality, such as wireless cyborg recharge. Their efficacy against mass radiation events has also dramatically increased.
```

### CODE REVIEW

**Comment by TobleroneSwordfish** [comment]
Not really sure why you'd ever build the zephyr, but the other ones look neat.

**Comment by Kubius** [comment]
Rough ideas I envisioned for how Zephyr might be used: disperse in high traffic hall areas for more time you can spend running, plop in medbay to keep ailing folks from passing out, or do some zesty augmented boxing. It's not something you'd want all the time, but it's also a good thing that it doesn't feel mandatory - can be a tricky balance to strike.

**Comment by TheTyrant25** [comment]
That applies to most of them, I think, which is good. Plus the different boards have different assembly material costs iirc so there's that aspect too..?

File: code/obj/machinery/interdictor.dm [hunk 1]
```diff
@@ -191,16 +231,26 @@
 //call this from an event to determine if sufficient power is remaining to complete an interdiction,
 //passing an amount in cell charge that is required to interdict the event.
 //returns 1 if interdiction was successful, 0 if power was insufficient
-//second arg skips immediate visual update (use if potential for very high amounts of individual calls)
-/obj/machinery/interdictor/proc/expend_interdict(var/stopcost,var/skipanim)
-	if (status & BROKEN || !src.canInterdict)
+
+//first arg specifies the cell charge required to successfully Do the Thing
```
Comment (flappybatpal) [inline code/obj/machinery/interdictor.dm:235]: As a proc should this comment not be in codedoc format to make it more useful when someone is writing a proc call?

**Review by flappybatpal — CHANGES_REQUESTED** [review]

File: code/datums/sims.dm [hunk 2]
```diff
@@ -251,6 +251,10 @@
 				if (protection > 0)
 					protection--
 					return 0
+				// Devera-class interdictor: ionizes away the "bodily funk" that would otherwise accumulate
```
[the remark this answers is not in the document]
Comment (Kubius) [reply code/datums/sims.dm]: Updated in 6ac706f.

File: code/mob/living/life/Life.dm [hunk 3]
```diff
@@ -259,6 +259,13 @@
 					animate(src, transform = matrix(), time = 1)
 				last_no_gravity = src.no_gravity
 
+			// Zephyr-class interdictor: creates an invigorating biofield in range
```
[the remark this answers is not in the document]
Comment (Kubius) [reply code/mob/living/life/Life.dm]: Updated in 6ac706f.

File: code/obj/machinery/interdictor.dm (hunk 1 above)
Comment (Kubius) [reply code/obj/machinery/interdictor.dm]: Rekerfuffled in 6ac706f; comment code-doc-ified, and updated a few of the variable names for a bit more clarity.

**Review by flappybatpal — APPROVED** [review]

**Pull request merged**